extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/CredentialsProviderConfig.java

```diff
@@ -13,13 +13,51 @@
     /**
      * Database credentials role, as defined by https://www.vaultproject.io/docs/secrets/databases/index.html
      *
-     * One of `database-credentials-role` or `kv-path` needs to be defined. not both.
```

Comment:
I haven't been able to find where it is enforced, please add a check

Reply:
`VaultCredentialsProvider` line 36 ish starts if statements that configure based on the first defined value.